drivers/sensor/pmw3360/pmw3360.c

```diff
@@ -790,21 +790,21 @@ static int pmw3360_sample_fetch(struct device *dev, enum sensor_channel chan)
 	int err = motion_burst_read(dev_data, data, sizeof(data));
 
 	if (!err) {
-		s32_t x = sys_get_le16(&data[PMW3360_DX_POS]);
-		s32_t y = sys_get_le16(&data[PMW3360_DY_POS]);
+		s32_t x = (s16_t)sys_get_le16(&data[PMW3360_DX_POS]);
```

Comment:
why not cast to s32_t ?

Reply:
`sys_get_le16` returns `u16_t` that I need to cast to `s16_t` so sign is be properly handled during the indirect conversion to `s32_t`.

We have not seen a problem here as we later on convert the value back to s16_t.

Reply:
ok

Reply:
> `sys_get_le16` returns `u16_t` that I need to cast to `s16_t` so sign is be properly handled during the indirect conversion to `s32_t`.
> 
> We have not seen a problem here as we later on convert the value back to s16_t.

Why not use `s16_t` for `x` and `y` then? These variables are only used in assignments to fields in `dev_data` that are `s16_t`.

Reply:
will check it

Reply:
Fixed